objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestProcessor.java

```diff
@@ -0,0 +1,57 @@
+package io.vantiq.extsrc.objectRecognition.neuralNet;
+
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vantiq.extsrc.objectRecognition.exception.ImageProcessingException;
+
+public class TestProcessor implements NeuralNetInterface2 {
+    
+    Logger log = LoggerFactory.getLogger(this.getClass());
+    int sleepTime = 1000;
+
+    @Override
+    public void setupImageProcessing(Map<String, ?> neuralNetConfig, String sourceName, String modelDirectory,
+            String authToken, String server) throws Exception {
+        if (neuralNetConfig.get("sleepTime") instanceof Integer && (Integer) neuralNetConfig.get("sleepTime") > 0) {
+            sleepTime = (Integer) neuralNetConfig.get("sleepTime");
+        }
+    }
+
+    @Override
+    public NeuralNetResults processImage(byte[] image) throws ImageProcessingException {
+        processImage(null, image);
+        return new NeuralNetResults();
+    }
+
+    @Override
+    public NeuralNetResults processImage(byte[] image, Map<String, ?> request) throws ImageProcessingException {
+        processImage(null, image, request);
+        return new NeuralNetResults();
+    }
+
+    @Override
+    public NeuralNetResults processImage(Map<String, ?> processingParams, byte[] image) throws ImageProcessingException {
+        try {
+            Thread.sleep(sleepTime);
```

Comment:
what's the purpose of the sleeps?  We should document what's going on in this class.

Reply:
I'll add documentation in this class to explain what is going on. The idea was that to ensure the the DiscardOldestPolicy was working as intended, we could use this class to mimic processing an image, but we would know exactly how long the "image processing" would take. 

This helped with writing a test to make sure that the maxRunningThreads/QueuedTasks and then DiscardOldestPolicy were both working as expected.